stat/stat.go

```diff
@@ -1001,6 +1005,42 @@ func Quantile(p float64, c CumulantKind, x, weights []float64) float64 {
 			}
 		}
 		panic("impossible")
+	case AvgDiscont:
```

Comment:
With this addition, I think it may be worth factoring the functions out into helpers.

Reply:
done.